server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodesTests.java

```diff
@@ -149,6 +150,18 @@ public void testResolveNodesIds() {
         assertThat(resolvedNodesIds, equalTo(expectedNodesIds));
     }
 
+    public void testMastersFirst() {
+        final List<DiscoveryNode> inputNodes = randomNodes(10);
+        final DiscoveryNodes.Builder discoBuilder = DiscoveryNodes.builder();
+        inputNodes.forEach(discoBuilder::add);
+        final List<DiscoveryNode> returnedNodes = discoBuilder.build().mastersFirstStream().collect(Collectors.toList());
+        assertEquals(returnedNodes.size(), inputNodes.size());
+        assertEquals(new HashSet<>(returnedNodes), new HashSet<>(inputNodes));
+        final List<DiscoveryNode> sortedNodes = new ArrayList<>(returnedNodes);
```

Comment:
Can we just sort the `inputNodes` and assert that this is what is returned?

Reply:
no, because the returned nodes will not be in insertion order (as an ImmutableOpenMap is used by DiscoveryNodes) after sorting them by masterness.